### PR TITLE
[TE] front end - Fix filter bar bugs

### PR DESCRIPTION
**Pull request by ttbach**
- Initialize filteredUrns on page load
- Added documentation for filter-bar-input component
- Added way to cache filtered results

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/components/filter-bar-input/component.js [hunk 1]
```diff
@@ -36,18 +60,29 @@ export default Ember.Component.extend({
...
 
       if (onSelect) {
-        const urns = Object.keys(entities).filter(urn => {
-          if (entities[urn].attributes[labelMapping]) {
-            return selectedValue.some(value => entities[urn].attributes[labelMapping].includes(value));
-          }
-        });
+        let urns;
+        // If there are no filters, show all entities under that event type
+        if (selectedValue.length == 0) {
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/components/filter-bar-input/component.js]: nit: `!selectedValue.length` is more concise

File: thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js [hunk 2]
```diff
@@ -62,6 +60,16 @@ export default Ember.Controller.extend({
...
+  filteredUrns: Ember.computed(
+    'entities',
+    function () {
+      const { entities } = this.getProperties('entities');
+      console.log("entities in filtered urns: ", entities);
+      return new Set(Object.keys(entities).filter(urn => entities[urn].type == 'event'
+        && entities[urn].eventType == 'holiday'));
+    }
+  ),
+
```
Comment (apucher) [inline thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js]: this shouldn't be here. filteredUrns is a set that's populated via actions

File: thirdeye/thirdeye-frontend/app/pods/rootcause/route.js [hunk 3]
```diff
@@ -18,7 +18,6 @@ export default Ember.Route.extend({
         'thirdeye:metric:194592', 'frontend:baseline:metric:194592',
         'thirdeye:event:holiday:2712391']),
       invisibleUrns: new Set(),
-      filteredUrns: new Set(),
```
Comment (apucher) [inline thirdeye/thirdeye-frontend/app/pods/rootcause/route.js]: see above

File: thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js [hunk 4]
```diff
@@ -14,8 +14,6 @@ export default Ember.Controller.extend({
 
   selectedUrns: null, // Set
 
-  filteredUrns: null, // Set
-
```
Comment (apucher) [inline thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js]: see below

File: thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js [hunk 5]
```diff
@@ -62,6 +60,16 @@ export default Ember.Controller.extend({
     }
   ),
 
+  filteredUrns: Ember.computed(
+    'entities',
+    function () {
+      const { entities } = this.getProperties('entities');
+      console.log("entities in filtered urns: ", entities);
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js]: is this still needed for debugging purposes?

Comment (ttbach) [reply thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js]: No, you're right. I forgot about this. Thanks

**Pull request merged**